control/statefbk.py

```diff
@@ -1119,18 +1119,23 @@ def gram(sys, type):
     if type not in ['c', 'o', 'cf', 'of']:
         raise ValueError("That type is not supported!")
 
-    # TODO: Check for continuous or discrete, only continuous supported for now
-        # if isCont():
-        #    dico = 'C'
-        # elif isDisc():
-        #    dico = 'D'
-        # else:
-    dico = 'C'
-
-    # TODO: Check system is stable, perhaps a utility in ctrlutil.py
-    # or a method of the StateSpace class?
-    if np.any(np.linalg.eigvals(sys.A).real >= 0.0):
-        raise ValueError("Oops, the system is unstable!")
+    # Check for continuous or discrete
+    if sys.isctime():
+        dico = 'C'
+
+        # TODO: Check system is stable, perhaps a utility in ctrlutil.py
+        # or a method of the StateSpace class?
+        if np.any(np.linalg.eigvals(sys.A).real >= 0.0):
+            raise ValueError("Oops, the system is unstable!")
+
+    elif sys.isdtime():
+        dico = 'D'
+
+        if np.any(np.abs(sys.poles()) >= 1.):
+            raise ValueError("Oops, the system is unstable!")
+
+    else:
+        raise ValueError("sys")
```

Comment:
This is true.  The only time this would ever get raised is if someone passed an object that returns false to both `isctime` and `isdtime`.  I put it in because it's bad practice to do `if cond1: ... elif cond2:` with no else but it would only realistically get raised during development if something was wrong.  That's why I left the message very simple because no real user is likely to ever see it, hopefully!

Reply:
If `sys.isdtime` literally returned `not sys.isctime` then we can be sure this will never occur.  But it looks more complicated than that.

Reply:
Okay, this may be pedantic but for the sake of coverage, please create a test so that the test suite at least hits that line once.